packages/zen-bull-netting/src/ZenBullNetting.sol

```diff
@@ -345,9 +466,58 @@ contract ZenBullNetting is Ownable, EIP712 {
         return sum;
     }
 
+    /**
+     * @notice get a withdraw receipt by index
+     * @param _index withdraw index in withdraws array
+     * @return receipt sender, amount and timestamp
+     */
     function getWithdrawReceipt(uint256 _index) external view returns (address, uint256, uint256) {
         Receipt memory receipt = withdraws[_index];
 
         return (receipt.sender, receipt.amount, receipt.timestamp);
     }
+
+    /**
+     * @notice get ZenBull token price using uniswap TWAP
+     * @return ZenBull price
+     */
+    function getZenBullPrice() external view returns (uint256) {
+        return _getZenBullPrice();
+    }
+
+    /**
+     * @dev check that the proposed ZenBull price is within a tolerance of the current Uniswap TWAP
+     */
+    function _checkZenBullPrice(uint256 _price) internal view {
+        uint256 zenBullFairPrice = _getZenBullPrice();
+
+        require(_price <= (zenBullFairPrice * (1e18 + otcPriceTolerance)) / 1e18, "ZBN12");
```

Comment:
not sure If I remember this right. did @aleone want two different tolerances for OTC price (auction clearing) and for netting?.